spec/cybersource/base_plugin_spec.rb

```diff
@@ -215,43 +292,97 @@ def with_transaction(kb_payment_id, transaction_type, created_at, context)
     t.destroy! unless t.nil?
   end
 
-  def purchase_with_card(expected_status = :PROCESSED, properties = [], expected_params = {})
+  def add_card_property(properties, card_type = :visa)
     properties << build_property('email', 'foo@bar.com')
-    properties << build_property('cc_number', '4111111111111111')
-
-    purchase(expected_status, properties, expected_params)
+    if card_type == :amex
+      properties << build_property('cc_number', '378282246310005')
+    else
+      properties << build_property('cc_number', '4111111111111111')
+    end
   end
 
-  def purchase_with_token(expected_status = :PROCESSED, properties = [], expected_params = {})
+  def add_token_property(properties)
     properties << build_property('email', 'foo@bar.com')
     properties << build_property('token', '1234')
-
-    purchase(expected_status, properties, expected_params)
   end
 
-  def purchase_with_network_tokenization(expected_status = :PROCESSED, properties = [], expected_params = {})
+  def add_network_tokenization_properties(properties, card_type = :visa)
+    if card_type == :amex
+      properties << build_property('cc_number', '378282246310005')
+      properties << build_property('brand', 'american_express')
+    else
+      properties << build_property('cc_number', '4111111111111111')
+      properties << build_property('brand', 'visa')
+    end
     properties << build_property('email', 'foo@bar.com')
-    properties << build_property('cc_number', '4111111111111111')
-    properties << build_property('brand', 'visa')
     properties << build_property('eci', '05')
-    properties << build_property('payment_cryptogram', '111111111100cryptogram')
+    properties << build_property('payment_cryptogram', Base64.encode64('111111111100cryptogram'))
```

Comment:
According to https://github.com/killbill/killbill-cybersource-plugin/blob/master/lib/cybersource/ext/active_merchant/active_merchant.rb#L63
Found there are some difference between amex and other cards, so for amex, there is a more decode step.
Code updated in the next commit.